net: Drop incoming packet if there is no data in it

jukkar opened the pull request:
If the network driver for some reason did not set the data in
the network packet properly, then just drop it as we cannot do
anything with just plain net_pkt.

Fixes #28131

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
net: Drop incoming packet if there is no data in it

If the network driver for some reason did not set the data in
the network packet properly, then just drop it as we cannot do
anything with just plain net_pkt.

Fixes #28131

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

tbursztyka commented on subsys/net/ip/net_core.c:
```diff
@@ -382,7 +382,7 @@ int net_recv_data(struct net_if *iface, struct net_pkt *pkt)
 	}
 
-	if (!pkt->frags) {
+	if (!pkt->frags || !pkt->frags->data) {
```
I think it's time to add a net_pkt_is_empty() function :)

jukkar replied:
That is a good idea, I will send a new version.

jukkar committed:
net: pkt: Add net_pkt_is_empty() util function

The function will check that the net_pkt will contain some
data and return true in that case, and false if the net_pkt
is empty.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

mniestroj commented on include/net/net_pkt.h:
```diff
@@ -900,6 +900,12 @@ static inline uint8_t *net_pkt_ip_data(struct net_pkt *pkt)
+static inline bool net_pkt_is_empty(struct net_pkt *pkt)
+{
+	return (!pkt->buffer || !net_pkt_data(pkt) ||
+		!pkt->buffer->len) ? true : false;
```
I don't think this `? true : false` is needed here.

jukkar replied:
Indeed, good point.

pfalcon replied:
Maybe use old good arithmetic comparison, `pkt->buffer->len == 0`? Friendly to humans, and even required by MISRA C.

jukkar replied:
Well, sure, that can be done too.
